odoo/addons/base/models/decimal_precision.py

```diff
@@ -25,9 +25,29 @@ class DecimalPrecision(models.Model):
         ('name_uniq', 'unique (name)', """Only one value can be defined for each given usage!"""),
     ]
 
+    @api.onchange('digits')
+    def _onchange_digits(self):
+        new_rounding = 1.0 / 10.0**self.digits
+        dangerous_uom = self.env['uom.uom'].search([('rounding', '<', new_rounding)])
```

Comment:
:warning: the module `uom` is not necessarily installed!
You should add this `onchange` method in module `uom` by inheriting `decimal.precision`.

Reply:
By the way, the warning only makes sense if you modify the digits of "Product Unit of Measure".
It does not make sense on "Product price" or "Payroll".